src/storage/src/hummock/sstable/block.rs

```diff
@@ -320,15 +502,18 @@ impl BlockBuilder {
 
     /// Calculate block size without compression.
     pub fn uncompressed_block_size(&mut self) -> usize {
-        self.buf.len() + (self.restart_points.len() + 1) * std::mem::size_of::<u32>()
+        self.buf.len()
+            + (self.restart_points.len() + 1) * std::mem::size_of::<u32>()
+            + (4 + 1) * self.restart_points_type_index.len()
+            + 4
     }
 
     /// Finishes building block.
     ///
     /// # Format
     ///
     /// ```plain
-    /// compressed: | entries | restart point 0 (4B) | ... | restart point N-1 (4B) | N (4B) |
+    /// compressed: | entries | restart point 0 (4B) | ... | restart point N-1 (4B) | N (4B) | restart point index 0 (5B)| ... | restart point index N-1 (5B) | N (4B)
```

Comment:
The doc here is not accurate. The implementation encodes all the index keys first and then all the index values.

Reply:
thanks ,  I decided to simplify the encode / decode code according to this format